### PR TITLE
Auto push to pypi

### PR DESCRIPTION
**Pull request by JohannesLusk**
This workflow will automatically push to pypi if a commit has a tag. The auto tag generation and pushing doesn't work as of yet.

### CODE REVIEW

**Review by Jegp — CHANGES_REQUESTED** [review]
Nice work! I enjoy reading the commit log :P

Does this work with the tagging? If so, I guess we should push to the "real" pypi and not the test one? 
Also, should `ChangeVersionNumber.py` be in `.gitignore`?

File: .gitignore [hunk 1]
```diff
@@ -4,4 +4,6 @@
 __pycache__/
 *.whl
 *.dat
-*.mp4
\ No newline at end of file
+*.mp4
+ChangeVersionNumber.py
```
Comment (Jegp) [inline .gitignore]: Shouldn't this line be kept out? I assume you'll want to **keep** the `ChangeVersionNumber.py` file, no?

**Review by Jegp — CHANGES_REQUESTED** [review]
Nice. LGTM! The only oddity I see is the addition of your change version number Python script into .gitignore. But once you change from `test.pypi.org` to the official repo, I'm happy to merge!

File: .gitignore (hunk 1 above)
Comment (JohannesLusk) [reply .gitignore]: Alrighty, Will do so, I tried to figure out the no start on push issue, gave up, and restructured it into two jobs instead. This should have the same effect with a lot less headache for me. See last commit

**Review by Jegp — CHANGES_REQUESTED** [review]
Ok, I think this is definitely progress, but it's not good if you run end up recursing too much. We really have to understand that before we can merge this. One possible approach could be to use this piece of code: https://github.com/marketplace/actions/github-tag

However, I _really_ think we should start with *just* the releasing. Meaning, just focusing on the push to PyPi when a new release is **manually** made. After that I'm happy to focus on making neat version number changes. But we can always automate that later.

**Review by Jegp — APPROVED** [review]
LGTM! :partying_face:

**Pull request merged**